xsd/pointwise_models/zscore.py

```diff
@@ -0,0 +1,243 @@
+import pandas as pd
+import numpy as np
+import xarray as xr
+
+from sklearn.base import RegressorMixin
+from sklearn.linear_model.base import LinearModel
+
+
+class ZScoreRegressor(LinearModel, RegressorMixin):
+    """ Z Score Regressor bias correction model wrapper
+
+    Apply a scikit-learn model (e.g. Pipeline) point-by-point. The pipeline
+    must implement the fit and predict methods.
+    """
+
+    def __init__(self, window_width=31, var_str='foo'):
+
+        self.window_width = window_width
+        self.var_str = var_str
+
+
+    def fit(self, X, y):
+        """ Fit Z-Score Model finds the shift and scale parameters
+        to inform bias correction.
+             
+        Parameters
+        ----------
+        X : pd.Series or pd.DataFrame, shape (n_samples, 1)
+            Training historical model data
+        y : pd.Series or pd.DataFrame, shape (n_samples, 1)
+            Target measured values.
+        window_width :  The size of the moving window for 
+            statistical analysis. Default is 31 days.
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+
+        X_mean, X_std = _calc_stats(X, self.window_width)
+        y_mean, y_std  = _calc_stats(y, self.window_width)
+        
+        shift, scale = _get_params(X_mean, X_std, y_mean, y_std)
+
+        self.shift = shift
+        self.scale = scale
+        
+        return self    
+        
+        
+    def predict(self, X):
+        """ Predict performs the z-score bias correction 
+        on the future model dataset, X.
+        
+        Parameters
+        ----------
+        X : DataFrame, shape (n_samples, 1)
+            Samples.
+        window_width : The size of the moving window for 
+            statistical analysis. Default is 31 days.
+        var_str :  The key associated with the target 
+            dataframe variable
+
+        Returns
+        -------
+        fut_corrected : pd.DataFrame, shape (n_samples, 1)
+            Returns corrected values.
+        """
+
+        fut_mean, fut_std, fut_zscore = _get_fut_stats(X, self.window_width)
+        shift_expanded, scale_expanded = _expand_params(X, self.var_str, self.shift, self.scale)
+
+        fut_mean_corrected, fut_std_corrected = _correct_fut_stats(fut_mean, fut_std, self.var_str, shift_expanded, scale_expanded)
+        fut_corrected = (fut_zscore * fut_std_corrected) + fut_mean_corrected
+        return fut_corrected
+  
+
+def _reshape(ds, window_width):
+    """
+    Helper function for `fit` that splits the year and day
+    dimensions of the time-coordinate and bookends the years
+    e.g. (Dec15:31 + whole year + Jan1:15) if window_width is 31 days.
+    ---------------------
+    Parameters:
+    ds : Xarray Dataset
+    window_width: the size of the rolling window.
```

Comment:
Thanks for doing a great job documenting these functions! Can you update the docstring format to follow the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) standard a bit more, e.g.: 

```Python
    """
    Helper function for `fit` that splits the year and day
    dimensions of the time-coordinate and bookends the years
    e.g. (Dec15:31 + whole year + Jan1:15) if window_width is 31 days.

    Parameters
    ----------
    ds : xr.Dataset
        A description of the dataset (structure)
    window_width : int
        the size of the rolling window.
...
```

Reply:
I did my best to follow the formatting (adding types and structure being the largest change). Let me know if I am not seeing something about the format that you'd like me to match.